unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+#include "ivi-wm-client-protocol.h"
+#include "ivi-input-client-protocol.h"
+
+extern "C"{
+WL_EXPORT const struct wl_interface ivi_screenshot_interface = {
+	"ivi_screenshot", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface ivi_wm_screen_interface = {
+	"ivi_wm_screen", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface ivi_wm_interface = {
+	"ivi_wm", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface ivi_input_interface = {
+	"ivi_input", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface wl_registry_interface = {
+	"wl_registry", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface wl_buffer_interface = {
+	"wl_buffer", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface wl_shm_pool_interface = {
+	"wl_shm_pool", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface wl_shm_interface = {
+	"wl_shm", 1,
+	0, NULL,
+	0, NULL,
+};
+
+WL_EXPORT const struct wl_interface wl_output_interface = {
+	"wl_output", 1,
+	0, NULL,
+	0, NULL,
+};
+
+FAKE_VALUE_FUNC(int, save_as_png, const char *, const char *, int32_t , int32_t , uint32_t );
+FAKE_VALUE_FUNC(int, save_as_bitmap, const char *, const char *, int32_t , int32_t , uint32_t );
+
+struct screenshot_data_t {
+    std::atomic<int32_t> fd;
+    std::atomic<uint32_t> error;
+    ilmErrorTypes result = ILM_SUCCESS;
+};
+
+}
+
+extern "C"{
+#include "ilm_control_wayland_platform.c"
+}
+
+static constexpr uint8_t MAX_NUMBER = 5;
+
+enum ilmControlStatus
+{
+    CREATE_LAYER    = 0,
+    DESTROY_LAYER   = 1,
+    CREATE_SURFACE  = 2,
+    DESTROY_SURFACE = 3,
+    NONE = 4
+};
+
+static ilmControlStatus g_ilmControlStatus = NONE;
+static t_ilm_notification_mask g_ilm_notification_mask;
+
+static void notificationCallback(ilmObjectType object, t_ilm_uint id, t_ilm_bool created, void *user_data)
+{
+    if (object == ILM_SURFACE)
+    {
+        g_ilmControlStatus = created ? CREATE_SURFACE : DESTROY_SURFACE;
+    } else if (object == ILM_LAYER)
+    {
+        g_ilmControlStatus = created ? CREATE_LAYER : DESTROY_LAYER;
+    }
+}
+
+static void surfaceCallbackFunction(t_ilm_surface surface, struct ilmSurfaceProperties* surfaceProperties, t_ilm_notification_mask mask)
+{
+    g_ilm_notification_mask = mask;
+    std::cout << "Notification: surface " << surface << "\n";
+}
+
+static void layerCallbackFunction(t_ilm_layer layer, struct ilmLayerProperties* layerProperties, t_ilm_notification_mask mask)
+{
+    g_ilm_notification_mask = mask;
+    std::cout << "Notification: layer " << layer << "\n";
+}
+
+class IlmControlTest : public ::testing::Test
+{
+public:
+    void SetUp()
+    {
+        init_ctx_list_content();
+        CLIENT_API_FAKE_LIST(RESET_FAKE);
+        g_ilmControlStatus = NONE;
+        g_ilm_notification_mask = ILM_NOTIFICATION_ALL;
+    }
+
+    void TearDown()
+    {
+        deinit_ctx_list_content();
+    }
+
+    void init_ctx_list_content()
```

Comment:
consider to divide this function to sub-functions, for supporting add new surface, seat, layer, and screen. it will be cleaner than currently.